vector/src/main/java/im/vector/app/core/pushers/UnifiedPushHelper.kt

```diff
@@ -150,31 +147,32 @@ class UnifiedPushHelper @Inject constructor(
                 .setTitle(stringProvider.getString(R.string.unifiedpush_getdistributors_dialog_title))
                 .setItems(distributorsName.toTypedArray()) { _, which ->
                     val distributor = distributors[which]
-                    if (distributor == UnifiedPush.getDistributor(context)) {
-                        Timber.d("Same distributor selected again, no action")
-                        return@setItems
-                    }
 
                     activity.lifecycleScope.launch {
-                        if (unregisterFirst) {
-                            // Un-register first
-                            unregister(pushersManager)
-                        }
                         UnifiedPush.saveDistributor(context, distributor)
                         Timber.i("Saving distributor: $distributor")
                         UnifiedPush.registerApp(context)
                         onDoneRunnable?.run()
                     }
                 }
-                .setCancelable(cancellable)
+                .setOnCancelListener {
```

Comment:
@Florian14 do you know whether this Listener was missing in Tchap? I'm asking because it was already present in Element-android  (I mean this Listener doesn't take part of the diff we cherry pick here according to [this](https://github.com/vector-im/element-android/pull/7085/files#diff-d0af9539e5621c00bfa7f4315721ab35c43af91f06e81d8f8a8b097d51a45487R157))

I just want to be sure we want it in Tchap. I think we need it

Reply:
This listener has been added by [this commit](https://github.com/vector-im/element-android/commit/0a2ce698becd39abddc85ca4263ea57a211ca052) from [this PR](https://github.com/vector-im/element-android/pull/6890) is Element-v1.4.34 (which is more recent than the last rebased version in Tchap, that's why this code was not present in Tchap).
We should probably also cherry-pick this PR for the fix.